### PR TITLE
Update to CSI v1.1.0

### PR DESCRIPTION
**Pull request by wongma7**
**Is this a bug fix or adding new feature?** new feature

**What is this PR about? / Why do we need it?** fixes https://github.com/kubernetes-sigs/aws-fsx-csi-driver/issues/39
- NodePublish request VolumeAttributes -> VolumeContext
- ValidateVolumeCapabilitiesResponse is different
- ControllerExpandVolume, NodeGetVolumeStats, and NodeExpandVolume are unimplemented.

**What testing is done?** ~WIP: it builds...however, tests still need to be updated. And ValidateVolumeCapabilities also needs to be implemented fully.~
All tests pass.

### CODE REVIEW

File: pkg/driver/controller.go [hunk 1]
```diff
@@ -166,10 +166,20 @@ func (d *Driver) ValidateVolumeCapabilities(ctx context.Context, req *csi.Valida
...
-		Supported: supported,
-	}, nil
+	confirmed := d.isValidVolumeCapabilities(volCaps)
+	if confirmed {
+		return &csi.ValidateVolumeCapabilitiesResponse{
+			Confirmed: &csi.ValidateVolumeCapabilitiesResponse_Confirmed{
+				// TODO if volume context is provided, should validate it too
+				// VolumeContext:      req.GetVolumeContext(),
+				VolumeCapabilities: volCaps,
+				// TODO if parameters are provided, should validate them too
```
Comment (wongma7) [inline pkg/driver/controller.go]: I have left 2 TODOs here for a follow-up PR. CreateVolume could use the same validation

File: tests/sanity/sanity_test.go [hunk 2]
```diff
@@ -53,6 +53,7 @@ var _ = AfterSuite(func() {
 })
 
 var _ = Describe("AWS FSx for Lustre CSI Driver", func() {
+	_ = os.MkdirAll("/tmp/csi", os.ModePerm)
```
Comment (leakingtapan) [inline tests/sanity/sanity_test.go]: Why is this needed?

Comment (wongma7) [reply tests/sanity/sanity_test.go]: sanity used to MkdirAll the paths for us, but now it only does Mkdir causing tests to fail without this (locally and in travis): https://github.com/kubernetes-csi/csi-test/commit/267dbbcb874cf12332ec966f372a10ad8df246d6#diff-cdf6b6b472a77c21dd6f192b6f3241eaL262

**Comment by onur-sam-gtn-ai** [comment]
README is not updated and this change broke all our EKS clusters relying on FSx suddenly.
Master version is no longer CSI 0.3. Our manifests were pointing to the latest tag and that broke the driver overnight for us.

**Comment by wongma7** [comment]
@onur-sam-gtn-ai sorry about that, I've opened https://github.com/kubernetes-sigs/aws-fsx-csi-driver/pull/76 to update the README. Please use the v0.1.0 tag as the latest tag is unstable, especially while the driver is alpha. (Beta is coming soon). Thanks.
